src/screens/DiscoveryScreenV2/hooks/useDiscovery.js

```diff
@@ -110,17 +138,39 @@ const useDiscovery = () => {
       setRefreshing(false);
     }
   };
+
+  const getIsMeFollowingTargetStatus = React.useCallback(
+    (userId) => {
+      const targetUser = discovery?.initialUsers?.find((item) => item?.user_id === userId);
+      const targetUserFromFollowed = discovery?.followedUsers?.find(
+        (item) => item?.user_id === userId
+      );
+      const targetUserFromFollowing = discovery?.unfollowedUsers?.find(
+        (item) => item?.user_id === userId
+      );
+
+      return (
+        targetUser?.is_followed ||
+        targetUserFromFollowed?.is_followed ||
+        targetUserFromFollowing?.is_followed ||
+        false
+      );
+    },
+    [discovery]
+  );
+
   return {
-    onRefreshDiscovery,
     refreshing,
-    updateFollowDiscoveryContext,
-    mapUser,
-    exhangeFollower,
+    topics,
     users,
-    updateFollowTopicDiscoveryContext,
+
+    exchangeFollower,
+    getIsMeFollowingTargetStatus,
     mapTopic,
-    topics,
-    topicExchangeFollower
+    onRefreshDiscovery,
+    topicExchangeFollower,
+    updateFollowDiscoveryContext,
+    updateFollowTopicDiscoveryContext
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The return statement now includes the new `getIsMeFollowingTargetStatus` function and the renamed `exchangeFollower` function. This is a good change as it allows these functions to be used elsewhere in the application, improving reusability.


<!-- This is an auto-generated comment by OSS CodeRabbit -->